pkg/controller/replication/replication_controller.go

```diff
@@ -237,6 +237,38 @@ func (rm *ReplicationManager) Run(workers int, stopCh <-chan struct{}) {
 // TODO: Surface that we are ignoring multiple controllers for a single pod.
 // TODO: use ownerReference.Controller to determine if the rc controls the pod.
 func (rm *ReplicationManager) getPodController(pod *api.Pod) *api.ReplicationController {
+	if rm.garbageCollectorEnabled {
+		controllerRef := controller.GetControllerOf(&pod.ObjectMeta)
```

Comment:
Thanks for doing this. I guess the improvement is not as big as 30774, given we already had a rc lookup cache. So let's merge 30774 first.